manifests/0000_30_cluster-api_13_vap.yaml

```diff
@@ -0,0 +1,121 @@
+
+    # annotation maps
+    - name: newAnn
+      expression: "object.metadata.?annotations.hasValue() ? object.metadata.?annotations.value() : {}"
+    - name: oldAnn
+      expression: "oldObject.metadata.?annotations.hasValue() ? oldObject.metadata.?annotations.value() : {}"
+
+    - name: specLockedExceptAuthoritativeAPI
+      expression: >
+        object.spec.authoritativeAPI != oldObject.spec.authoritativeAPI &&
+        [
+          [object.spec.?lifecycleHooks,   oldObject.spec.?lifecycleHooks],
+          [object.spec.?metadata,         oldObject.spec.?metadata],
+          [object.spec.?providerID,       oldObject.spec.?providerID],
+          [object.spec.?providerSpec,     oldObject.spec.?providerSpec],
+          [object.spec.?taints,           oldObject.spec.?taints]
+        ].all(p,
+          p[0].hasValue()
+            ? p[1].hasValue() && p[0].value() == p[1].value()
+            : !p[1].hasValue()
+        )
+
+  # All validations must evaluate to TRUE
+  validations:
+    # Only spec.authoritativeAPI may change
+    - expression: "object.spec == oldObject.spec || variables.specLockedExceptAuthoritativeAPI"
+      message:  "You may only modify spec.authoritativeAPI. Any other change inside .spec is not allowed."
+
+    # Guard machine.openshift.io/* and kubernetes.io/* labels
+    - expression: >
+        !(
+          variables.newLabels.exists(k,
+              (k.startsWith('machine.openshift.io') || k.startsWith('kubernetes.io')) &&
+              ((k in variables.oldLabels ? variables.oldLabels[k] : null) != variables.newLabels[k])
+          ) ||
+          variables.oldLabels.exists(k,
+              (k.startsWith('machine.openshift.io') || k.startsWith('kubernetes.io')) &&
+              !(k in variables.newLabels)
+          )
+        )
+      message: "Cannot modify or delete any machine.openshift.io/* or kubernetes.io/* label."
+
+    # Guard machine.openshift.io/* annotations
+    - expression: >
+        !(
+          variables.newAnn.exists(k,
+              k.startsWith('machine.openshift.io') &&
+              ((k in variables.oldAnn ? variables.oldAnn[k] : null) != variables.newAnn[k])
+          ) ||
+          variables.oldAnn.exists(k,
+              k.startsWith('machine.openshift.io') &&
+              !(k in variables.newAnn)
+          )
+        )
+      message: "Cannot modify or delete any machine.openshift.io/* annotation."
+
+    # Param-controlled labels (labels on the CAPI machine) may change only to match the value on the CAPI Machine
+    - expression: >
+        !variables.paramLabels.exists(k,
+          ((k in variables.newLabels ? variables.newLabels[k] : null) !=
+           (k in variables.oldLabels ? variables.oldLabels[k] : null))
+          &&
+          ((k in variables.newLabels ? variables.newLabels[k] : null) !=
+           variables.paramLabels[k])
+        )
+      message: "Cannot modify a Cluster API controlled label except to match the Cluster API mirrored machine."
```

Comment:
Hm not sure I follow here.

Reply:
So this is enforcing: `Are not being modified to not match the Authoritative resource`

Which I've interpreted as: `Are not being modified to not match the Authoritative resource === Are being modified away from matching Authoritative`

Reply:
Could you provide an example as well, I think it would help me understand better 😄

Reply:
Sorry - lost this! 

e.g 

CAPI (param) has labels `"foo":"bar"`, this would allow our MAPI to set `"foo":"bar"`, or to change from whatever was previously set to `"foo":"bar"`, but it would disallow any deviation from whatever is set on the CAPI label, so moving to `"foo":"baz"` would be denied. 

So we disallow common labels across machines from deviating - no shifting away from authority.

Reply:
Ok got it thanks!